src/core/caching/futures.ts

```diff
@@ -1,8 +1,40 @@
 import { RemoteItem, RemoteList } from 'utils/storeUtils';
 
+/**
+ * Encapsulates the state of an asynchronous operation to fetch a piece of data.
+ * Most commonly, this means a network request.
+ *
+ * ```mermaid
+ * flowchart TD
+ *   A[data: null\nerror: null\nisLoading: false]
+ *   B[data: null\nerror: null\nisLoading: true]
+ *   C[data: null\nerror: Error\nisLoading: false]
+ *   D[data: Object\nerror: null\nisLoading: false]
+ *   E{Success?}
+ *   A -->|Asynchronous operation begins| B
+ *   B -->|Asynchronous operation ends| E
+ *   E -->|Yes| D
+ *   E -->|No| C
+ * ```
```

Comment:
I'm guessing this should render a flowchart as some sort of graphic, but I'm not sure how to run the code in a way that does. I tried `yarn typedoc` which did build an output into `./typedoc/` that looked like something that I should be able to serve as static files, but that didn't work. What is the right way to view this?

Reply:
An update. It seems that it was just really slow to load (note the loading duration in screenshot below), and one file failed:

![image](https://github.com/user-attachments/assets/b1cba4ff-4ce5-4e06-8aa1-5d1aa607f68b)

After more than a minute of loading mermaid-related JS files, it finally rendered this:

![image](https://github.com/user-attachments/assets/c3079c39-dd99-465a-9067-0055f65be5de)

Is this something that you are also experiencing or is it just me?

Reply:
Yeah not been very impressed with the reliability of typescript-plugin-mermaid tbh